defaults/initrd.scripts

```diff
@@ -1966,15 +2043,27 @@ openLUKS() {
 							[ -e /dev/tty ] && run mv /dev/tty /dev/tty.org
 							run mknod /dev/tty c 5 1
 							cryptsetup_options="${cryptsetup_options} -d -"
-							gpg_cmd="gpg --logger-file /dev/null --quiet --decrypt ${mntkey}${LUKS_KEY} |"
+							gpg_cmd="gpg --logger-file /dev/null --quiet --decrypt ${mntkey}${LUKS_KEY} | "
+							ply_cmd="gpg --logger-file /dev/null --quiet --passphrase-fd 0 --batch --no-tty --decrypt ${mntkey}${LUKS_KEY} | "
 						fi
 					else
 						cryptsetup_options="${cryptsetup_options} -d ${mntkey}${LUKS_KEY}"
 					fi
 				fi
 				# At this point, keyfile or not, we're ready!
-				crypt_filter "${gpg_cmd}cryptsetup ${cryptsetup_options} luksOpen ${LUKS_DEVICE} ${LUKS_NAME}"
+				if [ "${PLYMOUTH}" = '1' ]
+				then
+					ply_cmd="${ply_cmd}cryptsetup ${cryptsetup_options} luksOpen ${LUKS_DEVICE} ${LUKS_NAME}"
```

Comment:
can you document what this is doing? I'm surprised to see plymouth changes touching LUKS

Reply:
Plymouth doesn't actually touch anything; but the actual cryptsetup command to unlock the LUKS partition is called by plymouth.  To do this, the command is built out (ply_cmd; the gpg call requires specifying stdin and not the standard password box) and written to a file.  This file is then read by plymouth to run the command (called in crypt_filter), pass the typed password from plymouth via stdin and unlock the LUKS partition.

Reply:
Should I provide commented documentation in the code?

Reply:
I've included documentation on the specific gpg options used in my new commit (squashed).